pkg/util/oauth/provider.go

```diff
@@ -0,0 +1,223 @@
+package oauth
```

Comment:
I think this file is not utils for common invoking, and should not put it in `util` folder. @wzqnls  WDYT?

Reply:
I think it would be better to put in `pkg/scm/provider`.